reactor-netty-http/src/main/java/reactor/netty/http/server/DefaultHttpServerRoutes.java

```diff
@@ -78,11 +80,23 @@ public HttpServerRoutes route(Predicate<? super HttpServerRequest> condition,
 		if (condition instanceof HttpPredicate) {
 			handlers.add(new HttpRouteHandler(condition,
 					handler,
-					(HttpPredicate) condition));
+					(HttpPredicate) condition, ((HttpPredicate) condition).uri));
 		}
 		else {
-			handlers.add(new HttpRouteHandler(condition, handler, null));
+			handlers.add(new HttpRouteHandler(condition, handler, null, null));
 		}
+
+		if (this.comparator != null) {
+			handlers.sort(this.comparator);
+		}
+
+		return this;
+	}
+
+	@Override
+	public HttpServerRoutes comparator(Supplier<Comparator<HttpRouteHandler>> supplier) {
```

Comment:
is there a huge benefit in requiring a `Supplier`?

Reply:
no,I have change method to `comparator(Comparator<HttpRouteHandlerMetadata> comparator)`